modin/pandas/series.py

```diff
@@ -495,7 +495,14 @@ def apply(self, func, convert_dtype=True, args=(), **kwds):
                 )
             ).__name__
         except Exception:
-            return_type = self.__name__
+            try:
+                return_type = type(
+                    getattr(pandas.Series([0], index=self.index[:1]), apply_func)(
```

Comment:
Will this work if the data is not numeric? Won't it throw an error if the data is string values and the `apply_func` assumes strings, but it is applied to an integer?

Reply:
Not numeric values were handled separately above, but now all the cases are handled by using the next expression `test_data = type(self[0])() if len(self.index) > 0 else []`.